File: src/Beginner/Day02/tip.py
def main():
    print("Welcome to the tip calculator!")
    while True:
        total_bill = input("What was your total bill?\n$")
        tip_percent = input("What percentage would you like to tip?\n")
        people_num = input("How many people are going to be splitting the bill?\n")

        try:
            total_bill = float(total_bill)
            tip_percent = float(tip_percent)
            people_num = int(people_num)
            break
        except (TypeError, ValueError) as e:
            print("Please enter numbers for all values")
            print(e)

    tipped_total = calc_tip(total_bill, tip_percent)
    print(f"Your total factoring in a tip is ${tipped_total:.2f}")
    if people_num > 1:
        print(
            f"Split between {people_num} people, each person would pay ${tipped_total / people_num:.2f}"
        )


def calc_tip(total: float, tip: float):
    multiplier = 1 + (tip / 100)
    return total * multiplier


if __name__ == "__main__":
    main()

File: 100-days-of-code/Beginner/Day02/src/tip.rs
use std::io;

pub fn tip_prompt() -> io::Result<(f64, f64)> {
    println!("Welcome to the tip calculator!");
    let mut ans = String::new();

    let total_bill = parse_input("What was your total bill?", &mut ans)?;
    let tip_percent = parse_input("What percentage would you like to tip?", &mut ans)?;
    let people_num = loop {
        match parse_input(
            "How many people are going to be splitting the bill?",
            &mut ans,
        ) {
            Ok(num) => {
                if num == num as usize as f64 && num > 0.0 {
                    break num;
                } else {
                    eprintln!("Please enter a positive whole number.")
                }
            }
            Err(err) => eprintln!("{}", err),
        }
    };

    let tipped_total = calc_tip(total_bill, tip_percent);

    Ok((tipped_total, people_num))
}

fn parse_input(prompt: &str, input: &mut String) -> io::Result<f64> {
    let res = loop {
        input.clear();
        println!("{}", prompt);
        io::stdin().read_line(input)?;
        match input.trim().parse::<f64>() {
            Ok(num) => break num,
            Err(err) => eprintln!("{}", err),
        }
    };
    Ok(res)
}

fn calc_tip(total: f64, tip: f64) -> f64 {
    let multiplier = 1 as f64 + (tip / 100 as f64);
    return total * multiplier;
}
